src/solver/simulation/sim_calcul_economique.cpp

```diff
@@ -319,16 +319,25 @@ void SIM_InitialisationProblemeHebdo(Data::Study& study,
 }
 
 void preparerBindingConstraint(const PROBLEME_HEBDO &problem, uint numSpace, int PasDeTempsDebut,
-                               const BindingConstraintsRepository &bindingConstraints, const uint weekFirstDay, int pasDeTemps) 
+                               const BindingConstraintsRepository &bindingConstraints,
+                               const BindingConstraintGroupRepository &bcgroup,
+                               const uint weekFirstDay, int pasDeTemps)
 {
     auto activeContraints = bindingConstraints.activeContraints();
     const auto constraintCount = activeContraints.size();
     for (unsigned constraintIndex = 0; constraintIndex != constraintCount; ++constraintIndex)
     {
         auto bc = activeContraints[constraintIndex];
         assert(bc->RHSTimeSeries().width && "Invalid constraint data width");
+        uint tmpts = 0;
+        for (const auto& group : bcgroup)
```

Comment:
You can get the group with:
auto* group = bcgroup[bc->group()]
if (group != nullptr) {
 ...
}